packages/headless/src/controllers/insight/facets/facet/headless-insight-facet.ts

```diff
@@ -0,0 +1,173 @@
+import {updateFacetOptions} from '../../../../features/facet-options/facet-options-actions';
+import {InsightEngine} from '../../../../app/insight-engine/insight-engine';
+
+export type {
+  FacetOptions,
+  FacetSearchOptions,
+  FacetValueState,
+  FacetProps,
+  Facet,
+  FacetState,
+  FacetSearch,
+  FacetSearchState,
+  SpecificFacetSearchResult,
+  FacetValue,
+  CoreFacet,
+  CoreFacetState,
+};
+
+/**
+ * Creates an `InsightFacet` controller instance.
+ *
+ * @param engine - The headless engine.
+ * @param props - The configurable `InsightFacet` properties.
+ * @returns An `InsightFacet` controller instance.
+ * */
+export function buildInsightFacet(
+  engine: InsightEngine,
+  props: FacetProps<FacetOptions>
+): Facet {
+  if (!loadFacetReducers(engine)) {
+    throw loadReducerError;
+  }
+
+  const {dispatch} = engine;
+  const coreController = buildCoreFacet(
+    engine,
+    {
+      ...props,
+      options: {
+        ...props.options,
+        ...(props.options.allowedValues && {
+          allowedValues: {
+            type: 'simple',
+            values: props.options.allowedValues,
+          },
+        }),
+      },
+    },
+    facetOptionsSchema
+  );
+  const getFacetId = () => coreController.state.facetId;
+
+  const createFacetSearch = () => {
+    const {facetSearch} = props.options;
+
+    return buildFacetSearch(engine, {
+      options: {facetId: getFacetId(), ...facetSearch},
+      select: (value) => {
+        dispatch(updateFacetOptions({freezeFacetOrder: true}));
+        dispatch(
+          executeSearch(
+            logFacetSelect({facetId: getFacetId(), facetValue: value.rawValue})
+          )
+        );
+      },
+    });
+  };
+
+  const facetSearch = createFacetSearch();
+  const {state, ...restOfFacetSearch} = facetSearch;
+
+  return {
+    ...coreController,
+
+    facetSearch: restOfFacetSearch,
+
+    toggleSelect(selection) {
+      coreController.toggleSelect(selection);
+      dispatch(
+        executeSearch(
+          getAnalyticsActionForToggleFacetSelect(getFacetId(), selection)
+        )
+      );
+    },
+
+    deselectAll() {
+      coreController.deselectAll();
+      dispatch(executeSearch(logFacetClearAll(getFacetId())));
+    },
+
+    sortBy(criterion: FacetSortCriterion) {
+      coreController.sortBy(criterion);
+      dispatch(
+        executeSearch(logFacetUpdateSort({facetId: getFacetId(), criterion}))
+      );
+    },
+
+    isSortedBy(criterion: FacetSortCriterion) {
+      return this.state.sortCriterion === criterion;
+    },
+
+    showMoreValues() {
+      coreController.showMoreValues();
+      dispatch(fetchFacetValues(logFacetShowMore(getFacetId())));
+    },
+
+    showLessValues() {
+      coreController.showLessValues();
+      dispatch(fetchFacetValues(logFacetShowLess(getFacetId())));
+    },
+
+    get state() {
+      return {
+        ...coreController.state,
+        facetSearch: facetSearch.state,
```

Comment:
The facet search actions are no-op right now correct? So interacting with this instance would simply de nothing?

Reply:
Good point, I haven't tried enabling facet search through the Insight engine... I'll validate.

Reply:
Yeah sounds good top me this is a temporary situation anyway.